tcpproxy.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package testing
```

Comment:
It could, I suppose, but it's simple enough and with minimal enough deps that I think it lives OK here. Also, tcpproxy.Proxy stutters and I can't think of a better name for the type :)